src/optical_spi.c

```diff
@@ -123,6 +132,19 @@ void check_received_opt_data(uint8_t num_expected_bytes){
 
             // creates a CAN message, but we'll use print statements for now
             print("recieved data from well_info 0x%X: 0x%X", current_well_info, opt_spi_data);
+        
+            // create CAN message
+            uint8_t tx_msg[8] = { 0x00 };
+            tx_msg[0] = current_cmd;            // opcode
+            tx_msg[1] = current_well_info;      // field number
+            tx_msg[2] = 0x00;                   // status, 0x00 = ok
+            tx_msg[3] = 0x00;                   // unused
+            tx_msg[4] = (opt_spi_data >> 16) & 0xFF; // data=24 bits, MSB first, zeroes padded on end
+            tx_msg[5] = (opt_spi_data >> 8) & 0xFF;
+            tx_msg[6] = (opt_spi_data) & 0xFF;
+            tx_msg[7] = 0x00;
```

Comment:
The data should go in bytes 5-7 instead of 4-6 because OBC always reads the 4 bytes as a `uint32_t`